mat/doc.go

```diff
@@ -36,10 +36,25 @@
 //  zero.Copy(a)
 //
 // Receivers must be the correct size for the matrix operations, otherwise the
-// operation will panic. As a special case for convenience, a zero-sized matrix
+// operation will panic. As a special case for convenience, a zero-value matrix
 // will be modified to have the correct size, allocating data if necessary.
 //  var c mat.Dense // construct a new zero-sized matrix
-//  c.Mul(a, a)       // c is automatically adjusted to be 6×6
+//  c.Mul(a, a)     // c is automatically adjusted to be 6×6
+//
+// Zero-value of a matrix
+//
+// A zero-value matrix is either the Go language definition of a zero-value or
+// a matrix that has had its Reset method called. Matrix implementations provided
+// by the mat package that can be reset have an IsZero method that returns whether
+// the matrix is a zero-value matrix. Internally, a zero-value is represented by a
+// zero-sized matrix with a zero-length stride. So the following will all result
+// in a zero-value matrix.
+//  var a mat.Dense
```

Comment:
I would number the examples or bullet the examples so it's clear lines 2 and 3 are separate examples.